sdk/storage/azure-storage-datamovement/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.13)
+project(azure-storage-datamovement LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+option(FETCH_SOURCE_DEPS "build source dependencies" OFF)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
+
+include(AzureVcpkg)
+include(AzureVersion)
+include(AzureCodeCoverage)
+include(AzureTransportAdapters)
+include(AzureDoxygen)
+include(AzureGlobalCompileOptions)
+include(AzureConfigRTTI)
+include(AzureBuildTargetForCI)
+
+az_vcpkg_integrate()
+
+if(FETCH_SOURCE_DEPS)
+    set(AZ_ALL_LIBRARIES ON)
+    include(FolderList)
+    SetCompileOptions(STORAGE_DATAMOVEMENT)
+    GetFolderList(STORAGE_DATAMOVEMENT)
+    foreach(oneFolder IN LISTS BUILD_FOLDERS)
+        message("add folder ${oneFolder}")
+        add_subdirectory(${oneFolder} EXCLUDE_FROM_ALL)
+    endforeach()
+elseif(NOT AZ_ALL_LIBRARIES)
+  find_package(azure-storage-blobs-cpp "12.4.0" CONFIG QUIET)
+  if(NOT azure-storage-blobs-cpp_FOUND)
+    find_package(azure-storage-blobs-cpp "12.4.0" REQUIRED)
+  endif()
+endif()
+
+set(
+  AZURE_STORAGE_DATAMOVEMENT_HEADER
+  inc/azure/storage/datamovement/blob_folder.hpp
+  inc/azure/storage/datamovement/datamovement_options.hpp
+  inc/azure/storage/datamovement/dll_import_export.hpp
+  inc/azure/storage/datamovement/job_properties.hpp
+  inc/azure/storage/datamovement/rtti.hpp
+  inc/azure/storage/datamovement/storage_transfer_manager.hpp
+  src/private/package_version.hpp
+)
+
+set(
+  AZURE_STORAGE_DATAMOVEMENT_SOURCE
+  src/storage_transfer_manager.cpp
+)
+
+add_library(azure-storage-datamovement ${AZURE_STORAGE_DATAMOVEMENT_HEADER} ${AZURE_STORAGE_DATAMOVEMENT_SOURCE})
+create_per_service_target_build(storage azure-storage-datamovement)
+
+# make sure that users can consume the project as a library.
+add_library(Azure::azure-storage-datamovement ALIAS azure-storage-datamovement)
+
+target_include_directories(
+  azure-storage-datamovement
+    PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
+      $<INSTALL_INTERFACE:include>
+)
+
+target_link_libraries(azure-storage-datamovement PUBLIC Azure::azure-storage-blobs)
```

Comment:
We need to call find_package() first
just the way you did here: https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/storage/azure-storage-common/CMakeLists.txt#L102

This is not super important, but if some people has installed azure-storage-blobs in their system and now they just want to build storage-datamovement, they should be able to do it, and we should be able to find the required dependency installed in the system.

Reply:
It's already done at Line 35 and 37. 

```CMake
elseif(NOT AZ_ALL_LIBRARIES)
  find_package(azure-storage-blobs-cpp "12.4.0" CONFIG QUIET)
  if(NOT azure-storage-blobs-cpp_FOUND)
    find_package(azure-storage-blobs-cpp "12.4.0" REQUIRED)
  endif()
endif()
```

Did I miss something?

Reply:
I missed it. Thanks